New HC Docs

katieclay opened the pull request:
:clipboard: Add associated issues, tickets, docs URL here.

## Overview

Reorganizing and polishing the HC docs.

## Changes

- Moved the HC docs hidden in the dev docs > features > account abstraction to be main HC docs 
- Rewrote/restructured existing material
- Removed the older HC docs

## Testing

I didn't change any of the code blocks, which I assume have already been tested?

InoMurko commented on integration-tests/hardhat.config.ts:
```diff
@@ -5,7 +5,8 @@ import '@nomiclabs/hardhat-ethers'
 
-const enableGasReport = !!process.env.ENABLE_GAS_REPORT
+// const enableGasReport = !!process.env.ENABLE_GAS_REPORT
+const enableGasReport = false
```
why doesn't this work?

trangtran-enya replied:
Looks like the result from GasReport overwrite Timing result, which is used for future run.

InoMurko replied:
Have you tried things like:
https://github.com/mochajs/mocha/pull/1360
https://www.npmjs.com/package/mocha-multi-reporters
?

trangtran-enya commented on ops/up_local.sh:
```diff
@@ -35,30 +35,30 @@ if [[ $BUILD == 1 ]]; then
-    docker-compose build -- monitor
-    docker-compose build -- verifier
-    docker-compose build -- replica
+  docker-compose build --parallel --no-cache -- builder l2geth l1_chain
+  docker-compose build --parallel --no-cache -- deployer dtl batch_submitter relayer integration_tests
+  docker-compose build --no-cache -- boba_message-relayer-fast
+  docker-compose build --no-cache -- gas_oracle
+  docker-compose build --no-cache -- boba_deployer
+  docker-compose build --no-cache -- fraud-detector
+  docker-compose build --no-cache -- monitor
+  docker-compose build --no-cache -- verifier
+  docker-compose build --no-cache -- replica
```
Testcases in the beginning failed, so I tried to run with image built from latest code.

souradeep-das reviewed (approved):
lgtm

wsdt reviewed (approved):
looks great! 🔥

Boyuan-Chen commented on integration-tests/test/boba-api.spec.ts:
```diff
@@ -68,8 +68,8 @@ describe('Boba API Tests', async () => {
     })
 
-    describe('Mainnet', async () => {
-      it('{tag:boba} should swap BOBA for ETH', async () => {
```
Do we want to remove all tags?

InoMurko replied:
eventually 🗡️

InoMurko commented on integration-tests/test/boba-fee-payment.spec.ts:
```diff
@@ -1044,9 +1044,9 @@ describe('Boba Fee Payment Integration Tests', async () => {
       )
       await transferBackTx.wait()
-    })
+    }).retries(3)
```
flaky test or flaky infra!

katieclay committed:
Reorganized and polished the HC docs, first pass

katieclay committed:
Edited TOC

katieclay committed:
Small aesthetic changes

katieclay committed:
Small mermaid diagram change

mmontour1306 commented on .gitbook/contents/hc/4_deploy-smart-contract.md:
```diff
@@ -0,0 +1,65 @@
+# Deploy the Smart Contract
+
+With a few adjustments to the `deploy.py` script, we can easily deploy our newly created smart contract.
```
This page is out of date. The script is now named "deploy-local.py" and it calls other Foundry scripts in crates/types/contracts/hc_scripts/

katieclay committed:
First half content restructuring

mmontour1306 commented on .gitbook/contents/hc/1_write-handler.md:
```diff
@@ -0,0 +1,152 @@
+
+## Simple Example
+
+Let's begin with a simple example where the handler receives two numbers. It will perform both addition and subtraction on these numbers. If the result of the subtraction results in an overflow (i.e. the first number is greater than the second), the handler will respond with an underflow error.
```
This line should say that an error will be generated if there is an underflow in the subtraction, when "b" is larger than "a" in the (a-b) expression.

mmontour1306 commented on .gitbook/contents/hc/3_write-smart-contract.md:
```diff
@@ -0,0 +1,191 @@
+
+There is an opportunity for a `HybridAccount` contract to implement a billing system here, requiring a payment of `ERC20` tokens or some other mechanism of collecting payment from the calling contract. This is optional.
+
+## Implement a Helper Contract
```
This section is describing what happens within the system-wide HCHelper contract. Developers don't have to implement a Helper contract in this version of Hybrid Compute (although they did in the old system).

katieclay committed:
Step 3 revisions, small corrections per Michael

katieclay committed:
Small clarification

mmontour1306 commented on .gitbook/contents/hc/3_write-smart-contract.md:
```diff
@@ -0,0 +1,190 @@
+Now we can write the smart contract, which will call our previously created off-chain handler. Our contract has two purposes:
+
+1. Intentionally burn ETH gas (in order to test gas estimation logic).
+2. Count how many times the contract has been called based on certain inputs and errors.
```
This should be "increment a counter variable" rather than "count how many times"

mmontour1306 commented on .gitbook/contents/hc/3_write-smart-contract.md:
```diff
@@ -0,0 +1,190 @@
+1. Intentionally burn ETH gas (in order to test gas estimation logic).
+2. Count how many times the contract has been called based on certain inputs and errors.
+
+You can find the needed `HybridAccount` contract along with its dependencies in our [repository](https://github.com/bobanetwork/account-abstraction-hc/contracts/samples/HybridAccount.sol). Additionally, the contract is pulled in as a submodule when you checkout our [`rundler-hc` repository](https://github.com/bobanetwork/rundler-hc).
```
The submodule path is rundler-hc/crates/types/contracts/lib/account-abstraction/...

katieclay committed:
More Michael suggestion edits

wsdt commented on .gitbook/contents/hc/3_write-smart-contract.md:
````diff
@@ -0,0 +1,192 @@
+  event CalledFrom(address sender);
+```
+
+To implement our function to waste gas, we can leverage a `for` loop to increase the cost of transacting:
````
This is more for representation purposes rather than a requirement.

katieclay replied:
Wasting gas? Or the `for` loop?

wsdt commented on .gitbook/contents/hc/3_write-smart-contract.md:
```diff
@@ -0,0 +1,192 @@
+contract TestCounter {
+    mapping(address => uint256) public counters;
+
+    address payable immutable demoAddr;
```
```suggestion
    address payable immutable hcAccount;
```

wsdt commented on .gitbook/contents/hc/4_deploy-smart-contract.md:
````diff
@@ -0,0 +1,62 @@
+After loading the contract, we can deploy it as follows:
+
+```solidity
+epAddr = deploy2("EntryPoint", EP.constructor(), 0)
````
maybe lets add a note here that deploy2 is just a function that deploys the smart contract with the deploy2 opcode for deterministic smart contract addresses.

wsdt commented on .gitbook/contents/hc/4_deploy-smart-contract.md:
```diff
@@ -0,0 +1,62 @@
+    )
+
+    // Add Sub
+    server.register_function(offchain_addsub2, selector("addsub2(uint32,uint32)"))  # 97e0d7ba
```
each of these register_function statements represents a whitelisting for a offchain rpc server functionality (e.g. addSub example = 1 whitelisting).

wsdt reviewed (approved):
left some comments

katieclay committed:
Changes based on Kevin's suggestions

katieclay committed:
More changes for Kevin suggestions

katieclay committed:
Added link to deploy script, code block language to python referencing it

katieclay committed:
Removed explanation for older version of deployment

mmontour1306 committed:
Update, including new deploy-local.py changes (#239)

* Update some docs, including a description of the most recent changes
to the deploy-local.py script (to match sepolia-rc14 images).

 Changes to be committed:
	modified:   .gitbook/contents/hc/3_write-smart-contract.md
	modified:   .gitbook/contents/hc/4_deploy-smart-contract.md

* Update .gitbook/contents/hc/3_write-smart-contract.md

Co-authored-by: katieclay <111803175+katieclay@users.noreply.github.com>

* Update .gitbook/contents/hc/3_write-smart-contract.md

Co-authored-by: katieclay <111803175+katieclay@users.noreply.github.com>

* Update .gitbook/contents/hc/4_deploy-smart-contract.md

Co-authored-by: katieclay <111803175+katieclay@users.noreply.github.com>

* Update .gitbook/contents/hc/4_deploy-smart-contract.md

Co-authored-by: katieclay <111803175+katieclay@users.noreply.github.com>

* Update .gitbook/contents/hc/4_deploy-smart-contract.md

Co-authored-by: katieclay <111803175+katieclay@users.noreply.github.com>

* Update .gitbook/contents/hc/4_deploy-smart-contract.md

Co-authored-by: katieclay <111803175+katieclay@users.noreply.github.com>

* Update .gitbook/contents/hc/4_deploy-smart-contract.md

Co-authored-by: katieclay <111803175+katieclay@users.noreply.github.com>

* Update .gitbook/contents/hc/4_deploy-smart-contract.md

Co-authored-by: katieclay <111803175+katieclay@users.noreply.github.com>

* Update .gitbook/contents/hc/4_deploy-smart-contract.md

Co-authored-by: katieclay <111803175+katieclay@users.noreply.github.com>

* Update .gitbook/contents/hc/4_deploy-smart-contract.md

Co-authored-by: katieclay <111803175+katieclay@users.noreply.github.com>

---------

Co-authored-by: katieclay <111803175+katieclay@users.noreply.github.com>

mmontour1306 committed:
Merge branch 'develop' into katie/hc-docs
